coala_quickstart/info_extractors/GruntfileInfoExtractor.py

```diff
@@ -0,0 +1,275 @@
+        return the elements of type "Literal" by searching recursively.
+        """
+        if (obj and obj.get("type") == "ArrayExpression" and
+                "elements" in obj.keys()):
+            return [
+                elem["value"]
+                for elem in obj["elements"] if elem["type"] == "Literal"
+            ]
+
+        elif obj and obj.get("type") == "ObjectExpression":
+            if (obj.get("value") and
+                obj["value"]["type"] == "ArrayExpression" and
+                    "elements" in obj["value"].keys()):
+                return [
+                    elem["value"]
+                    for elem in obj["elements"] if elem["type"] == "Literal"
+                ]
+            else:
+                return self.extract_literals_from_expression(obj.get("value"))
+
+        elif obj and obj.get("type") == "Property":
+            return self.extract_literals_from_expression(obj.get("value"))
+
+    def get_task_config(self, parsed_config):
+        """
+        From the parsed configuration object for a task,
+        extract and return the config settings in the form
+        of a python dict.
+
+        :param parsed_config:
+            A portion of  `PyJsParser().parse` instance of
+            "Gruntfile.js" that contains configuration of
+            a task defined appearing in "Gruntfile.js" in the form
+            of `grunt.initConfig({'task': 'config'})`
+        :returns:
+            A python dictionary of configurations in the form of
+            key-value pairs.
+        """
+        result = {}
+        for prop in parsed_config["properties"]:
+            prop_value = None
+            if prop["value"]["type"] == "Identifier":
+                prop_value = prop["value"]["name"]
+
+            elif prop["value"]["type"] == "Literal":
+                prop_value = prop["value"]["value"]
+
+            elif prop["value"]["type"] == "ObjectExpression":
+                prop_value = self.get_task_config(prop["value"])
+
+            elif prop["value"]["type"] == "ArrayExpression":
+                prop_value = self.extract_literals_from_expression(prop)
+
+            else:
+                logging.warn(
+                    "quickstart was not able to parse the value "
+                    "of the config {}".format(
+                        prop["key"]["name"]))
+
+            result[prop["key"]["name"]] = prop_value
+
+        return result
+
+    def extract_globs(self, parsed_config):
+        """
+        Extract the path glob literals matching ceratin keys
+         from the PyJsParser().parse instance.
+        :param parsed_file:
+            An instance of PyJsParser().parse
+        :return:
+            A dictionary with keys `use` and `ignore` and values being the
+            corresponding list of globs.
+        """
+        to_use_keys = ['all', 'main', 'files', 'src', 'sources']
+        to_ignore_keys = ['ignore', 'exclude']
+
+        result = {
+            "include": [],
```

Comment:
The first one is prone to mistakes when ignore_globs and include_globs can be swapped and it could be hard to detect this. Not sure what should be the best way though :sweat_smile: 

For now changed the key names to include_globs and ignore_globs